neqo-transport/src/server.rs

```diff
@@ -383,11 +383,11 @@ impl Server {
         }
     }
 
-    fn create_qlog_trace(&self, attempt_key: &AttemptKey) -> NeqoQlog {
+    fn create_qlog_trace(&self, odcid: &ConnectionIdRef<'_>) -> NeqoQlog {
```

Comment:
`PublicPacket::dcid` returns a `&ConnectionIdRef<'a>`:

https://github.com/mozilla/neqo/blob/9e6cd932b140fde141e6414d53f539b6d296d4f9/neqo-transport/src/packet/mod.rs#L676-L678

Thus I opted for `create_qlog_trace` to take a `&ConnectionIdRef` instead of a `&ConnectionId` since the former can be derived from the latter for free.

Let me know if that is fine by you, or whether you consider this a premature optimization not worth the noise.

(I am a bit confused why `PublicPacket::dcid` returns a reference to a reference (i.e. `&` and `Ref<'_>`) to some data. But that might be out of scope for this pull request.)

Reply:
Oh, the ref to a ref is probably just to avoid putting `Copy` on that struct.  Which is silly because it is just a reference itself.